src/transaction/txParser.ts

```diff
@@ -183,34 +225,58 @@ const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
   )
 }
 
-const parseTxWithdrawals = (
-  withdrawals: any,
-): _Withdrawal[] => {
+const parseTxWithdrawals = (withdrawals: any): _Withdrawal[] => {
   if (!isWithdrawalsMap(withdrawals)) {
     throw Error(Errors.WithrawalsParseError)
   }
   return Array.from(withdrawals).map(([address, coins]): _Withdrawal => ({ address, coins: BigInt(coins) }))
 }
 
-const parseTtl = (ttl: any): number => {
-  if (ttl === null || ttl === undefined) {
-    throw Error(Errors.MssingTTLParseError)
+const parseFee = (fee: any): Lovelace => {
+  if (!isUint64(fee)) {
+    throw Error(Errors.FeeParseError)
+  }
+  return BigInt(fee)
+}
+
+const parseTtl = (ttl: any): BigInt | undefined => {
+  if (ttl && !isUint64(ttl)) {
+    throw Error(Errors.TTLParseError)
+  }
+  return ttl && BigInt(ttl)
+}
+
+const parseValidityIntervalStart = (validityIntervalStart: any): BigInt | undefined => {
+  if (validityIntervalStart && !isUint64(validityIntervalStart)) {
+    throw Error(Errors.ValidityIntervalStartParseError)
+  }
+  return validityIntervalStart && BigInt(validityIntervalStart)
+}
+
+const parseMetaDataHash = (metaDataHash: any): Buffer | undefined => {
+  if (metaDataHash && !Buffer.isBuffer(metaDataHash)) {
```

Comment:
even a numerical zero/empty string would make it through this parsing as zero is falsy in js

Instead this should be something like:

```
if (metadataHash != null && !Buffer.isBuffer(metaDataHash)
```

Reply:
Refactored